### PR TITLE
Adds support for fallback by handling failures in nextTimestamp (Issue #3467)

### PR DESCRIPTION
**Pull request by dkelkhoff**
With an implementation of nextTimestamp copied from hibernate-jache 5.2 Timestamper

Addresses https://github.com/redisson/redisson/issues/3467

### CODE REVIEW

File: redisson-hibernate/redisson-hibernate-52/src/main/java/org/redisson/hibernate/RedissonRegionFactory.java [hunk 1]
```diff
@@ -167,6 +179,28 @@ public long nextTimestamp() {
                 + "redis.call('set', KEYS[1], nextValue); "
                 + "return nextValue;",
                 RScript.ReturnType.INTEGER, Arrays.<Object>asList("redisson-hibernate-timestamp"), time);
+        } catch(Exception e) {
+            if(fallback) {
+                return fallbackNextTimestamp();
```
Comment (mrniko) [inline redisson-hibernate/redisson-hibernate-52/src/main/java/org/redisson/hibernate/RedissonRegionFactory.java:184]: use `SimpleTimestamper.next()` instead

**Review by mrniko — CHANGES_REQUESTED** [review]

File: redisson-hibernate/redisson-hibernate-52/src/main/java/org/redisson/hibernate/RedissonRegionFactory.java (hunk 1 above)
Comment (dkelkhoff) [reply redisson-hibernate/redisson-hibernate-52/src/main/java/org/redisson/hibernate/RedissonRegionFactory.java:184]: `SimpleTimestamper` isn't available in Hibernate 5.2.  

Hibernate 5.2 has this logic in `org.hibernate.cache.jcache.time.Timestampe`, under `hibernate-jcache`, but I'm assuming we wouldn't want to add `hibernate-jcache` as a dependency here, correct?  

Please advise - I'm glad to change, however you deem appropriate. 

Thanks!

Comment (mrniko) [reply redisson-hibernate/redisson-hibernate-52/src/main/java/org/redisson/hibernate/RedissonRegionFactory.java:184]: There is a problem - we can't get LGPL 2.1 code and put it in project under Apache License.

Comment (dkelkhoff) [reply redisson-hibernate/redisson-hibernate-52/src/main/java/org/redisson/hibernate/RedissonRegionFactory.java:184]: Can we add this dependency then, in redisson-hibernate-52's pom.xml?
```
       <dependency>
         <groupId>org.hibernate</groupId>
         <artifactId>hibernate-jcache</artifactId>
         <version>5.2.18.Final</version>
         <optional>true</optional>
       </dependency>
```

And then call `Timestamper.next()`?

Thanks.

Comment (mrniko) [reply redisson-hibernate/redisson-hibernate-52/src/main/java/org/redisson/hibernate/RedissonRegionFactory.java]: I need to think how to make it better

**Comment by mrniko** [comment]
Fixed in https://github.com/redisson/redisson/commit/34d1739f72868755fe659a7483df6a7ac23915ae
Thanks for your help!

**Pull request closed without merging**